### PR TITLE
cleanup code

### PR DESCRIPTION
**Pull request by dbu**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | => travis-ci
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/DependencyInjection/CmfRoutingExtension.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ private function setupFormTypes(array $config, ContainerBuilder $container, Load
     {
         $loader->load('form-type.xml');
 
-        if (isset($config['dynamic'])) {
+        if (array_key_exists('dynamic', $config)) {
```
Comment (wouterj) [inline src/DependencyInjection/CmfRoutingExtension.php]: not sure about this one. Isset seems to do the job very well.

File: src/Routing/DynamicRouter.php [hunk 2]
```diff
@@ -87,17 +87,17 @@ protected function cleanDefaults($defaults, Request $request = null)
...
         }
 
-        if (isset($defaults[RouteObjectInterface::CONTENT_OBJECT])) {
+        if (array_key_exists(RouteObjectInterface::CONTENT_OBJECT, $defaults)) {
             $request->attributes->set(self::CONTENT_KEY, $defaults[RouteObjectInterface::CONTENT_OBJECT]);
             unset($defaults[RouteObjectInterface::CONTENT_OBJECT]);
         }
 
-        if (isset($defaults[RouteObjectInterface::TEMPLATE_NAME])) {
+        if (array_key_exists(RouteObjectInterface::TEMPLATE_NAME, $defaults)) {
```
Comment (wouterj) [inline src/Routing/DynamicRouter.php]: same in this file

File: src/DependencyInjection/CmfRoutingExtension.php (hunk 1 above)
Comment (dbu) [inline src/DependencyInjection/CmfRoutingExtension.php]: the problem with isset is that the following code would not lead to an error:

```
function foo(array $config) 
{
    if (isset($konfig['dynamic'])) {
        ....
```

that is, typos in variable names get ignored. i found this to be quite a hassle in refactorings.

File: src/DependencyInjection/CmfRoutingExtension.php [hunk 3]
```diff
@@ -207,7 +207,7 @@ private function setupDynamicRouter(array $config, ContainerBuilder $container,
         $dynamic->replaceArgument(2, new Reference($config['url_generator']));
     }
 
-    private function loadPhpcrProvider($config, XmlFileLoader $loader, ContainerBuilder $container, array $locales, $matchImplicitLocale)
+    private function loadPhpcrProvider($config, LoaderInterface $loader, ContainerBuilder $container, array $locales, $matchImplicitLocale)
```
Comment (pamil) [inline src/DependencyInjection/CmfRoutingExtension.php]: `$config` -> `array $config`?

File: src/DependencyInjection/CmfRoutingExtension.php [hunk 4]
```diff
@@ -254,7 +242,7 @@ private function loadInitializer($config, XmlFileLoader $loader, ContainerBuilde
         $loader->load('initializer-phpcr.xml');
     }
 
-    private function loadOrmProvider($config, XmlFileLoader $loader, ContainerBuilder $container, $matchImplicitLocale)
+    private function loadOrmProvider($config, LoaderInterface $loader, ContainerBuilder $container, $matchImplicitLocale)
```
Comment (pamil) [inline src/DependencyInjection/CmfRoutingExtension.php]: `$config` -> `array $config`?

File: src/Doctrine/Phpcr/RedirectRoute.php [hunk 5]
```diff
@@ -60,6 +60,8 @@ class RedirectRoute extends RedirectRouteModel implements PrefixInterface, Hiera
      * Additional options:
      *
      * * add_trailing_slash: When set, a trailing slash is appended to the route
+     *
+     * @param array $options Options
```
Comment (pamil) [inline src/Doctrine/Phpcr/RedirectRoute.php]: I think we can omit the message if it duplicates variable name.

File: src/Model/Route.php [hunk 6]
```diff
@@ -233,13 +237,11 @@ public function getPath()
...
     {
-        $len = strlen($this->getStaticPrefix());
-
-        if (strncmp($this->getStaticPrefix(), $pattern, $len)) {
+        if (0 !== strpos($pattern, $this->getStaticPrefix())) {
             throw new \InvalidArgumentException('You can not set a pattern for the route that does not start with its current static prefix. First update the static prefix or directly use setVariablePattern.');
         }
 
-        return $this->setVariablePattern(substr($pattern, $len));
+        return $this->setVariablePattern(substr($pattern, $this->getStaticPrefix()));
```
Comment (pamil) [inline src/Model/Route.php]: `$this->getStaticPrefix()` returns string, `substr` expects int - I guess it should be `strlen($this->getStaticPrefix())`.

Comment (dbu) [inline src/Model/Route.php]: ups, good catch! thanks, fixed.

File: src/DependencyInjection/CmfRoutingExtension.php (hunk 3 above)
Comment (dbu) [inline src/DependencyInjection/CmfRoutingExtension.php]: fixed

File: src/DependencyInjection/CmfRoutingExtension.php (hunk 4 above)
Comment (dbu) [inline src/DependencyInjection/CmfRoutingExtension.php]: fixed

**Comment by dbu** [comment]
updated. @wouterj can you live with the array_key_exists change?

**Comment by wouterj** [comment]
Yes, no problem :)

There are a couple build failures, are these caused by these changes?

**Comment by wouterj** [comment]
@dbu if you require `doctrine/orm` 2.5.* (or just `doctrine/instantiator` `^2.5`), the lower deps build should be fixed as well.

**Comment by dbu** [comment]
woop woop! please review and hopefully merge!

**Comment by ElectricMaxxx** [comment]
LGTM.

**Pull request merged**